### PR TITLE
feat(base): added utils

### PR DESCRIPTION
**Pull request by e11sy**
- added keyboard util
- added caret util
- added dom util

### CODE REVIEW

File: packages/caret/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
+import $, { isCollapsedWhitespaces } from '@utils/dom';
```
Comment (neSpecc) [inline packages/caret/src/index.ts:1]: get rid of $, import needed functions separately

File: packages/dom/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,567 @@
+import * as _ from './utils';
```
Comment (neSpecc) [inline packages/dom/src/index.ts:1]: import needed functions separately

File: package.json [hunk 3]
```diff
@@ -1,21 +1,31 @@
 {
   "name": "utils",
+  "private": true,
```
[the remark this answers is not in the document]
Comment (e11sy) [reply package.json:3]: image
 https://yarnpkg.com/en/docs/cli/install

File: eslint.config.mjs [hunk 4]
```diff
@@ -1,25 +1,43 @@
...
+import { plugin as TsPlugin, parser as TsParser } from 'typescript-eslint';
 /**
  * @todo connect architecture config
  */
 export default [
   ...CodeX,
   {
-    name: 'ts-editorjs/utils',
-
+    name: 'ts-editorls/utils',
```
Comment (neSpecc) [inline eslint.config.mjs:10]:
```suggestion
    name: 'ts-editorjs/utils',
```

**Review by neSpecc — APPROVED** [review]

**Pull request merged**